python/lsst/daf/butler/registry/bridge/monolithic.py

```diff
@@ -168,23 +174,36 @@ def insert(self, refs: Iterable[DatasetIdRef]) -> None:
 
     def forget(self, refs: Iterable[DatasetIdRef]) -> None:
         # Docstring inherited from DatastoreRegistryBridge
-        rows = self._refsToRows(self.check(refs))
-        self._db.delete(self._tables.dataset_location, ["datastore_name", "dataset_id"], *rows)
+        with self._db.transaction():
+            for refs_chunk in chunk_iterable(refs, 50_000):
```

Comment:
What's the motivation for the chunking here? Can there be a comment explaining why? Is it because the performance goes out of control if the chunk is too large now that there is a deleteWhere instead of a simple delete?

Reply:
I am worried that with many millions parameters the query becomes too large. I saw errors in my case which looked like there is a limit somewhere on query size (at least with EXPLAIN). The previous code used `Database.delete` and `Database.insert` which do 1k-chunks internally (either in Database or in sqlalchemy). Here I'm constructing WHERE subquery myself, so I need to limit it to a reasonable size. I'll add a comment.